src/vs/workbench/parts/debug/browser/debugActions.ts

```diff
@@ -215,7 +215,12 @@ export class RestartAction extends AbstractDebugAction {
 	static LABEL = nls.localize('restartDebug', "Restart");
 	static RECONNECT_LABEL = nls.localize('reconnectDebug', "Reconnect");
 
-	constructor(id: string, label: string, @IDebugService debugService: IDebugService, @IKeybindingService keybindingService: IKeybindingService) {
+	constructor(id: string, label: string,
+		@IDebugService debugService: IDebugService,
+		@IKeybindingService keybindingService: IKeybindingService,
+		@IWorkspaceContextService private contextService?: IWorkspaceContextService,
```

Comment:
Why are these arguments optional?

Reply:
The bottom two arguments are there because we need those two services for starting a debugger.  They're passed by what seems to be reflection in src\vs\platform\instantiation\common\instantiationService.ts:120 (this is how StartAction is called also).  This the black magic that the other contributor mentioned in #39060. 

However, existing calls to RestartAction, such as at src\vs\workbench\parts\debug\browser\debugActionsWidget.ts:253 and src\vs\workbench\parts\debug\electron-browser\callStackView.ts:259 do not pass an IWorkspaceContextService or an IHistoryService, which is why we made them optional.  Let us know your thoughts on this.